### ISSUE TITLE
[Bug]: Shopify theme dev --store . An unexpected error occurred

### ISSUE THREAD

**Issue by rossthomson89** [Shopify/cli#2155]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Running "**shopify theme dev --store nameofstore.myshopify.com**" should log me into the Shopify CLI and allow me to run a local version at 127.0.0.1 

### Actual behavior

Running "**shopify theme dev --store nameofstore.myshopify.com**" throws an "An unexpected error occurred".

Stack Trace Shows:

/usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli/identity_auth.rb:141:in `server': uninitialized constant ShopifyCLI::IdentityAuth::Servlet (NameError)

        server.mount("/", Servlet, self, state_token)
                          ^^^^^^^
        from /usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/lib/shopify_cli/identity_auth.rb:154:in `block in initiate_authentication'

I've tried the suggestions here: https://github.com/Shopify/shopify-cli/issues/1640#issuecomment-1086541395 but to no avail... 

### Verbose output

```shell
shopify:theme:dev init version: @oclif/core@2.1.4 argv: [ '--store', 'halcyon-spirits.myshopify.com', '--verbose' ] +0ms
2023-06-12T20:23:48.419Z: Ensuring that the user is authenticated with the Theme API with the following scopes:
[]

2023-06-12T20:23:48.420Z: Ensuring that the user is authenticated with the Admin API with the following scopes for the store halcyon-spirits.myshopify.com:
[]

2023-06-12T20:23:48.421Z: Getting session store...
2023-06-12T20:23:48.426Z: Validating existing session against the scopes:
[
  "openid",
  "https://api.shopify.com/auth/shop.admin.graphql",
  "https://api.shopify.com/auth/shop.admin.themes",
  "https://api.shopify.com/auth/partners.collaborator-relationships.readonly",
  "https://api.shopify.com/auth/shop.storefront-renderer.devtools",
  "https://api.shopify.com/auth/partners.app.cli.access"
]
For applications:
{
  "adminApi": {
    "scopes": [],
    "storeFqdn": "halcyon-spirits.myshopify.com"
  }
}

2023-06-12T20:23:48.428Z: Sending Identity Introspection request to URL: https://accounts.shopify.com/oauth/introspection
2023-06-12T20:23:48.429Z: Sending POST request to URL https://accounts.shopify.com/oauth/introspection
With request headers:
 - User-Agent: Shopify CLI; v=3.46.4
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-12T20:23:48.695Z: Request to https://accounts.shopify.com/oauth/introspection completed in 252 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"8ad7bbb09082510acebe2d08908be2a3"
 - x-request-id: 8307bb0f-25e2-4c2a-80a7-5643f4ba6cec
    
2023-06-12T20:23:48.701Z: The identity token is valid: true
2023-06-12T20:23:48.702Z: 
The validation of the token for application/identity completed with the following results:
- It's expired: false
- It's invalid in identity: false
  
2023-06-12T20:23:48.702Z: The current session is valid but needs refresh. Refreshing...
2023-06-12T20:23:48.704Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=refresh_token&access_token=atkn_872b7e1fcb8fcdc835f60ac7a8545490f0a0a8bebd9a06d1f71c42c2a0d48c72&refresh_token=atkn_ea8181f2af714b544c4f7e557d3174d54bf6f8ef81f31f5c51d96a76dfc74bef&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3
With request headers:
 - User-Agent: Shopify CLI; v=3.46.4
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-12T20:23:49.078Z: Request to https://accounts.shopify.com/oauth/token?grant_type=refresh_token&access_token=atkn_872b7e1fcb8fcdc835f60ac7a8545490f0a0a8bebd9a06d1f71c42c2a0d48c72&refresh_token=atkn_ea8181f2af714b544c4f7e557d3174d54bf6f8ef81f31f5c51d96a76dfc74bef&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3 completed in 372 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"134c16d7fc6cf9998ba723bd180f7833"
 - x-request-id: 89d230c5-ed81-415c-a387-ea014f0df714
    
2023-06-12T20:23:49.081Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=271e16d403dfa18082ffb3d197bd2b5f4479c3fc32736d69296829cbb28d41a6&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.app.cli.access&subject_token=atkn_aecce6aaa47af0badf039d9b82ac30d01ba9472dc25ce65de33a3461f895db70
With request headers:
 - User-Agent: Shopify CLI; v=3.46.4
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-12T20:23:49.082Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=ee139b3d-5861-4d45-b387-1bc3ada7811c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.storefront-renderer.devtools&subject_token=atkn_aecce6aaa47af0badf039d9b82ac30d01ba9472dc25ce65de33a3461f895db70
With request headers:
 - User-Agent: Shopify CLI; v=3.46.4
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-12T20:23:49.383Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=271e16d403dfa18082ffb3d197bd2b5f4479c3fc32736d69296829cbb28d41a6&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.app.cli.access&subject_token=atkn_aecce6aaa47af0badf039d9b82ac30d01ba9472dc25ce65de33a3461f895db70 completed in 299 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"bb5741c02cdab2ad8ae772fa82747b01"
 - x-request-id: e1c915de-d916-47d0-b078-dc116c6ae85c
    
2023-06-12T20:23:49.544Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=ee139b3d-5861-4d45-b387-1bc3ada7811c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.storefront-renderer.devtools&subject_token=atkn_aecce6aaa47af0badf039d9b82ac30d01ba9472dc25ce65de33a3461f895db70 completed in 458 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"6ffac82992a46958fe6e40c16630f063"
 - x-request-id: 242f68f4-7079-4d03-8285-47d7595e47e3
    
2023-06-12T20:23:49.547Z: Sending POST request to URL https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=7ee65a63608843c577db8b23c4d7316ea0a01bd2f7594f8a9c06ea668c1b775c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.graphql+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.themes+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.collaborator-relationships.readonly&subject_token=atkn_aecce6aaa47af0badf039d9b82ac30d01ba9472dc25ce65de33a3461f895db70&destination=https%3A%2F%2Fhalcyon-spirits.myshopify.com%2Fadmin
With request headers:
 - User-Agent: Shopify CLI; v=3.46.4
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-12T20:23:49.795Z: Request to https://accounts.shopify.com/oauth/token?grant_type=urn%3Aietf%3Aparams%3Aoauth%3Agrant-type%3Atoken-exchange&requested_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&subject_token_type=urn%3Aietf%3Aparams%3Aoauth%3Atoken-type%3Aaccess_token&client_id=fbdb2649-e327-4907-8f67-908d24cfd7e3&audience=7ee65a63608843c577db8b23c4d7316ea0a01bd2f7594f8a9c06ea668c1b775c&scope=openid+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.graphql+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fshop.admin.themes+https%3A%2F%2Fapi.shopify.com%2Fauth%2Fpartners.collaborator-relationships.readonly&subject_token=atkn_aecce6aaa47af0badf039d9b82ac30d01ba9472dc25ce65de33a3461f895db70&destination=https%3A%2F%2Fhalcyon-spirits.myshopify.com%2Fadmin completed in 245 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"d1fb80c0a8c7f253b8a1b8cdd1b027f9"
 - x-request-id: 00837242-de37-4f88-b77c-74e3bb744140
    
2023-06-12T20:23:49.797Z: Setting session store...
2023-06-12T20:23:49.821Z: Ensuring that the user is authenticated with the Storefront API with the following scopes:
[]

2023-06-12T20:23:49.822Z: Getting session store...
2023-06-12T20:23:49.824Z: Validating existing session against the scopes:
[
  "openid",
  "https://api.shopify.com/auth/shop.admin.graphql",
  "https://api.shopify.com/auth/shop.admin.themes",
  "https://api.shopify.com/auth/partners.collaborator-relationships.readonly",
  "https://api.shopify.com/auth/shop.storefront-renderer.devtools",
  "https://api.shopify.com/auth/partners.app.cli.access"
]
For applications:
{
  "storefrontRendererApi": {
    "scopes": []
  }
}

2023-06-12T20:23:49.824Z: Sending Identity Introspection request to URL: https://accounts.shopify.com/oauth/introspection
2023-06-12T20:23:49.825Z: Sending POST request to URL https://accounts.shopify.com/oauth/introspection
With request headers:
 - User-Agent: Shopify CLI; v=3.46.4
 - Keep-Alive: timeout=30
 - Sec-CH-UA-PLATFORM: darwin
 - Content-Type: application/json

2023-06-12T20:23:50.035Z: Request to https://accounts.shopify.com/oauth/introspection completed in 208 ms
With response headers:
 - cache-control: no-cache, no-store, private, must-revalidate, max-age=0
 - content-type: application/json; charset=utf-8
 - etag: W/"7e60a57acdc27df90cd6b212b7529964"
 - x-request-id: 459f32e1-92e8-41d8-a51c-3c40626d5a95
    
2023-06-12T20:23:50.038Z: The identity token is valid: true
2023-06-12T20:23:50.038Z: 
The validation of the token for application/identity completed with the following results:
- It's expired: false
- It's invalid in identity: false
  
2023-06-12T20:23:50.039Z: Getting development theme...
2023-06-12T20:23:50.511Z: Request to https://halcyon-spirits.myshopify.com/admin/api/unstable/themes/122497237063.json?fields=id completed in 464 ms
With response headers:
 - content-type: application/json; charset=utf-8
 - x-request-id: b59d6b2d-cc5f-4b93-bcba-d718179867a6
    
╭─ success ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                                                                                                                          │
│  Preview your theme                                                                                                                                                                                                      │
│    • http://127.0.0.1:9292 [1]                                                                                                                                                                                           │
│                                                                                                                                                                                                                          │
│  Next steps                                                                                                                                                                                                              │
│    • Customize your theme at the theme editor [2]                                                                                                                                                                        │
│    • Share your theme preview [3]                                                                                                                                                                                        │
│                                                                                                                                                                                                                          │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
[1] http://127.0.0.1:9292
[2] https://halcyon-spirits.myshopify.com/admin/themes/122497237063/editor
[3] https://halcyon-spirits.myshopify.com/?preview_theme_id=122497237063

2023-06-12T20:23:50.604Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/ruby -v
  · Working directory: /Users/ross/Documents/Projects/halcyon-spirits

2023-06-12T20:23:50.624Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle -v
  · Working directory: /Users/ross/Documents/Projects/halcyon-spirits

2023-06-12T20:23:50.810Z: Reading the content of file at /usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/Gemfile...
2023-06-12T20:23:50.813Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle config set --local path /Users/ross/Library/Caches/shopify-gems-nodejs
  · Working directory: /usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-12T20:23:50.991Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle config set --local without development:test
  · Working directory: /usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-12T20:23:51.165Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle install
  · Working directory: /usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby

2023-06-12T20:23:51.513Z: 
Running system process:
  · Command: /usr/local/opt/ruby/bin/bundle exec /usr/local/opt/ruby/bin/ruby /usr/local/Cellar/shopify-cli/3.46.4/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify theme serve /Users/ross/Documents/Projects/halcyon-spirits --live-reload hot-reload --theme 122497237063
  · Working directory: /Users/ross/Documents/Projects/halcyon-spirits

✗ An unexpected error occurred.
        To submit an issue include the stack trace.
        To print the stack trace, add the environment variable SHOPIFY_CLI_STACKTRACE=1.
```


### Reproduction steps

1. Upgraded to the latest CLI version
2. Run "shopify theme dev --store nameofstore.myshopify.com"
3. When prompted to "login" via the browser the "Unexpected Error" Occurs 

### Operating System

Mac OS Ventura 13.3.1

### Shopify CLI version (check your project's `package.json` if you're not sure)

 3.46.4

### Shell

_No response_

### Node version (run `node -v` if you're not sure)

v20.3.0

### What language and version are you using in your application?

Ruby 3.2.2

**Comment by mgmanzella** [comment]
cc: @Shopify/theme-code-tools

**Comment by hannahgledhill** [comment]
I'm having exactly the same issue with same stack trace working from Windows 10 on version 3.46.4. Have also tried rolling back to previous minor releases of 3.46.x but am now just completely unable to log back in to the store and seeing the same error every time.

**Comment by gonzaloriestra** [comment]
There's server-side incident related to authentication that is probably causing this issue. We are investigating it, sorry for the inconvenience.

**Comment by MichaelBallenAcid** [comment]
I have the same problem with Shopify CLI version: 3.46.4 in MAC

**Comment by MaxDesignFR** [comment]
@MichaelBallenAcid They just released 3.46.5, it fixed the issue for me.
